core/src/main/scala/com/salesforce/op/stages/impl/feature/SmartTextVectorizer.scala

```diff
@@ -156,38 +165,59 @@ class SmartTextVectorizer[T <: Text](uid: String = UID[SmartTextVectorizer[T]])(
 }
 
 object SmartTextVectorizer {
-  val MaxCardinality = 100
+  val MaxCardinality: Int = 100
+  val MinTextLengthStdDev: Double = 0
   private[op] def partition[T: ClassTag](input: Array[T], condition: Array[Boolean]): (Array[T], Array[T]) = {
     val all = input.zip(condition)
-    (all.collect { case (item, true) => item }.toSeq.toArray, all.collect { case (item, false) => item }.toSeq.toArray)
+    (all.collect { case (item, true) => item }, all.collect { case (item, false) => item })
   }
 }
 
 /**
  * Summary statistics of a text feature
  *
- * @param valueCounts counts of feature values
+ * @param valueCounts  counts of feature values
+ * @param lengthCounts counts of token lengths
  */
-private[op] case class TextStats(valueCounts: Map[String, Int]) extends JsonLike
+private[op] case class TextStats(
+  valueCounts: Map[String, Int],
+  lengthCounts: Map[Int, Int]
+) extends JsonLike {
+
+  val lengthSize = lengthCounts.values.sum
+  val lengthMean: Double = lengthCounts.foldLeft(0.0)((acc, el) => acc + el._1 * el._2) / lengthSize
+  val lengthVariance: Double = lengthCounts.foldLeft(0.0)(
+    (acc, el) => acc + el._2 * (el._1 - lengthMean) * (el._1 - lengthMean)
+  )
+  val lengthStdDev: Double = math.sqrt(lengthVariance / lengthSize)
+}
 
 private[op] object TextStats {
   def monoid(maxCardinality: Int): Monoid[TextStats] = new Monoid[TextStats] {
     override def plus(l: TextStats, r: TextStats): TextStats = {
-      if (l.valueCounts.size > maxCardinality) l
-      else if (r.valueCounts.size > maxCardinality) r
-      else TextStats(l.valueCounts + r.valueCounts)
+      val newValueCounts = if (l.valueCounts.size > maxCardinality) l.valueCounts
+        else if (r.valueCounts.size > maxCardinality) r.valueCounts
+        else l.valueCounts + r.valueCounts
+
+      val newLengthCounts = if (l.lengthCounts.size > maxCardinality) l.lengthCounts
+        else if (r.lengthCounts.size > maxCardinality) r.lengthCounts
+        else l.lengthCounts + r.lengthCounts
```

Comment:
can we make this a function?

Reply:
Probably, there's a chunk for `Map[String, Int]` and a chunk for `Map[Int, Int]` so we should be able to combine since each has a monoid.